packages/aws-cdk-lib/core/lib/removal-policies.ts

```diff
@@ -0,0 +1,154 @@
+import { IConstruct } from 'constructs';
+import { Aspects, IAspect } from './aspect';
+import { CfnResource } from './cfn-resource';
+import { RemovalPolicy } from './removal-policy';
+
+/**
+ * Properties for applying a removal policy
+ */
+export interface RemovalPolicyProps {
```

Comment:
I think it would be good to add a property `priority` to `RemovalPolicyProps`? (ref: https://github.com/aws/aws-cdk/blob/v2.172.0/packages/aws-cdk-lib/core/lib/aspect.ts#L45)

(* The `priority` was introduced in v2.172.0: https://dev.to/aws-heroes/aws-cdk-aspects-specifications-have-changed-3i75.)

And the `apply` method will be changed like:

```ts
  public apply(policy: RemovalPolicy, props: RemovalPolicyProps = {}) {
    Aspects.of(this.scope).add(new RemovalPolicyAspect(policy, props), {
      priority: props.priority ?? AspectPriority.MUTATING,
    });
  }
```

Reply:
thanks. 
https://github.com/aws/aws-cdk/pull/32283/commits/dec21e2e6e12df733f5e2ef70dd98b871bab2722

Reply:
Thanks for the change.

> If multiple aspects apply conflicting settings, the one with the higher priority wins.

I found the words "the higher priority" and "wins" a little confusing.

The higher priority (lower numeric value) is applied first, and the lower priority (higher numeric value) is applied later. (Complicated explanation...)

In other words, the removal policy will be the value by the former if `overwrite` is false, and the value by the latter if true, right?

Can we explain that well?

Reply:
https://github.com/aws/aws-cdk/pull/32283#discussion_r1899120716